src/snosearch/fields.py

```diff
@@ -573,11 +575,16 @@ def render(self, *args, **kwargs):
         Returns constructed query in debug field if debug param specified.
         '''
         self.parent = kwargs.get('parent')
+        self.query_builder = AbstractQueryFactory(
+            params_parser=self.get_params_parser(),
+            **self.kwargs
+        )
         if self.get_params_parser().get_debug():
             self.response.update(
                 {
                     DEBUG_KEY: {
-                        RAW_QUERY: self.get_query_builder().search.to_dict()
+                        RAW_QUERY: self.get_query_builder().search.to_dict(),
+                        SPECIFIED_INDICES: self.query_builder._get_index()
```

Comment:
I think you can do `self.get_query_builder().search._index` to get the indices actually set on the Search object? https://github.com/opensearch-project/opensearch-dsl-py/blob/c80360685e260c730d483b5c394560771b328248/opensearch_dsl/search.py#L129